SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd fill constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+// For std::isnan
+#include <cmath>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+namespace esimd_tests = esimd_test::api::functional::ctors;
+namespace esimd_functional = esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_init =
+        esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_init.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in variable declaration
+// context.
+struct var_dec {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_var_decl(init_value, step);
+    simd_by_var_decl.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in rvalue in an
+// expression context.
+struct rval_in_express {
+  static std::string get_description() { return "rvalue in an expression"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_rval;
+    simd_by_rval = esimd::simd<DataT, NumElems>(init_value, step);
+    simd_by_rval.copy_to(out);
+  }
+};
+
+// Descriptor class for the case of calling constructor in const reference
+// context.
+class const_ref {
+public:
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    return call_simd_by_const_ref<DataT, NumElems>(
+        esimd::simd<DataT, NumElems>(init_value, step), out);
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Enumeration of possible values for base value and step that will be provided
+// into simd constructor.
+enum class init_val {
+  min,
```

Comment:
```suggestion
enum class init_val {
  none = 0,
  min,
```
Minor, just in general.
By using `= 0` we are stating the values of enum explicitly.
By providing some `none` value we have some `default` value to use for variable initialization and further failure trigger in case of some error in algorithm.

Reply:
I going to use this enum only in comparative statements, so we don't care about values of the fields